classes/requests/order/session/class-ledyer-create-hpp.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Create HPP
+ *
+ * @package Ledyer\Requests\Order\Session
+ */
+
+namespace Ledyer\Requests\Order\Session;
+
+use Ledyer\Requests\Order\Request_Order;
+
+defined( 'ABSPATH' ) || exit();
+
+/**
+ * Class Create_HPP
+ *
+ * @package Ledyer\Requests\Order\Session
+ */
+class Create_HPP extends Request_Order {
+	/**
+	 * Adds a confirmation URL to the cart data.
+	 *
+	 * @param array $data The WooCommerce cart data.
+	 * @param int   $order_id The WooCommerce order id.
+	 * @return array The modified cart data.
+	 */
+	public function set_confirmation_url( $data, $order_id ) {
+		$merchant_urls = ledyer()->merchant_urls->get_urls( $order_id );
+		if ( $merchant_urls ) {
+			$data['settings']['urls']['confirmation'] = $merchant_urls['confirmation'];
+		}
+		return $data;
+	}
+
+	/**
+	 * Creates the HPP URL.
+	 *
+	 * @param string $session_id The Ledyer Checkout session to use for the HPP request.
+	 * @return string The HPP URL.
+	 */
+	public function create_hpp_url( $session_id ) {
+		// Todo: remove hardcoding here.
+		return 'https://pos.sandbox.ledyer.com/?sessionId=' . $session_id;
```

Comment:
Where is the URL for production/live? The sandbox endpoint is only used for testing.

Reply:
I think that pos.live should be used for non testing. Should I add a check for test mode in this method?

Reply:
Yes, since we need to support both.